lib/beaker-hostgenerator/generator.rb

```diff
@@ -104,6 +112,19 @@ def __parse_node_info_token(token)
         node_info['arbitrary_roles'] = []
       end
 
+      if node_info['host_settings']
+        # Turns a string like "{foo=bar,this=that}" into a
+        # hash like {'foo' => 'bar', 'this' => 'that'}.
+        node_info['host_settings'] = Hash[
+          node_info['host_settings'].
+          delete('{}').
+          split(',').
+          map { |keyvalue| keyvalue.split('=') }
+        ]
+      else
+        node_info['host_settings'] = {}
+      end
+
       return node_info
     end
```

Comment:
Did you intend to comment on the `generate` method instead?  https://github.com/puppetlabs/beaker-hostgenerator/pull/28/files#diff-0623c137fe4358c047889a4ff1adc01bR20

I only say that because I was thinking the same thing, and that method is longer and harder to read than this one.
But I'm happy to decompose both! 😄